### PR TITLE
Make generated implicit remoting modules backwards compatible with PowerShell 5

### LINKED ISSUE #17132: Export-PSSession generated module does not load in PowerShell 5.1 even though ScriptGenerator version is the same
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

It works if done other way around (exported on PS5, imported on PS7).

Seems like at some point Export-PSSession moved to a different assembly. Considering that `VersionOfScriptGenerator` has stayed on 1.0, that seems to imply that generated modules should be backwards compatible. Changing:
https://github.com/PowerShell/PowerShell/blob/7cc9c874730e87b06dbe54d50e77c1c472996c54/src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs#L2112

to something like the following fixes the issue
```csharp
string versionOfScriptGenerator = "$executionContext.InvokeCommand.GetCommand('Export-PSSession', [System.Management.Automation.CommandTypes]::Cmdlet).ImplementingType::VersionOfScriptGenerator";
```

### Steps to reproduce

On PS7 export remote session to a module:
```powershell
$s =  New-PSSession -ComputerName TestMachine

Export-PSSession -Session $s -CommandName Get-Date -OutputModule $PWD\Remote_GetDate -AllowClobber
```

On PS 5.1 try to load the module
```powershell
Import-Module .\Remote_GetDate
```

### Expected behavior

```console
# Remote_GetDate module loads, can run the implicit remoting command
(Get-Date).PSComputerName
```

### Actual behavior

```console
# Module fails to load
Unable to find type [Microsoft.PowerShell.Commands.ExportPSSessionCommand,Microsoft.PowerShell.Commands.Utility, Version=7.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]
```

### Error details

```console
Unable to find type [Microsoft.PowerShell.Commands.ExportPSSessionCommand,Microsoft.PowerShell.Commands.Utility, Version=7.0.0.0, Culture=neutral, PublicKeyToken=31bf38
56ad364e35].
At D:\Dev\Remote_GetDate\Remote_GetDate.psm1:15 char:57
+ ... Generator = [Microsoft.PowerShell.Commands.ExportPSSessionCommand, Mi ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (Microsoft.Power...1bf3856ad364e35:TypeName) [], RuntimeException
    + FullyQualifiedErrorId : TypeNotFound

Import-Module : The module cannot be loaded because it has been generated with an incompatible version of the Export-PSSession cmdlet. Generate the module with the Expo
rt-PSSession cmdlet from the current session, and try loading the module again.
At line:1 char:1
+ Import-Module .\Remote_GetDate
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (The module cann...e module again.:String) [Import-Module], RuntimeException
    + FullyQualifiedErrorId : The module cannot be loaded because it has been generated with an incompatible version of the Export-PSSession cmdlet. Generate the modul
   e with the Export-PSSession cmdlet from the current session, and try loading the module again.,Microsoft.PowerShell.Commands.ImportModuleCommand
```

### Environment data

```powershell
# Module exported on
Name                           Value
----                           -----
PSVersion                      7.3.0-preview.3
PSEdition                      Core
GitCommitId                    7.3.0-preview.3-28-g7cc9c874730e87b06dbe54d50e77c1c472996c54
OS                             Microsoft Windows 10.0.22000
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by Tadas**
# PR Summary

Make generated implicit remoting modules backwards compatible with PowerShell 5 (Fix #17132)

## PR Context

Fixing this issue would allow for a better user experience in environments that haven't been fully migrated to a modern version of PowerShell yet.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

**Review by PaulHigin — APPROVED** [review]
This change replaces a string to be executed containing a hardcoded assembly name, with a different string expression that finds the correct type, all to access the script generator version property.  This looks like a good change to me.  But I would like @daxian-dbw to also take a look.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs
Comment (Tadas) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/ImplicitRemotingCommands.cs:2112]: Maybe I'm doing something wrong but downlevel powershell is unable to find type without a version: `[Microsoft.PowerShell.Commands.ExportPSSessionCommand, Microsoft.PowerShell.Commands.Utility,  Culture=neutral, PublicKeyToken=31bf3856ad364e35]`.

It works with `[Microsoft.PowerShell.Commands.ExportPSSessionCommand, Microsoft.PowerShell.Commands.Utility]` or `[System.Reflection.Assembly]::LoadWithPartialName('Microsoft.PowerShell.Commands.Utility, Culture=neutral, PublicKeyToken=31bf3856ad364e35')` but I wouldn't be able to say which is the correct approach here.

**Comment by Tadas** [comment]
Sorry @daxian-dbw, is there anything else for me to do here?

**Comment by daxian-dbw** [comment]
@Tadas As said in my last comment https://github.com/PowerShell/PowerShell/pull/17227#discussion_r908736348, I prefer `[Microsoft.PowerShell.Commands.ExportPSSessionCommand, Microsoft.PowerShell.Commands.Utility]` as in
```c#
Type type = typeof(ExportPSSessionCommand);
string asmName = type.Assembly.GetName().Name;
string versionOfScriptGenerator = $"[{type.FullName}, {asmName}]::VersionOfScriptGenerator";
```
But this is not a blocking comment, so it's up to @PaulHigin to decide if merging this PR as is.

**Comment by PaulHigin** [comment]
@daxian-dbw  Can you sign off please?

**Comment by daxian-dbw** [comment]
@PaulHigin I have a concern as described in https://github.com/PowerShell/PowerShell/pull/17227#discussion_r868289143 and prefer to be explicit about the expected assembly name. But it's not a blocking comment.

**Comment by PaulHigin** [comment]
@daxian-dbw   It is a blocking comment if you don't sign off :).  I won't merge until I get two sign offs.  If you feel you cannot sign off then please work with the author to remedy.

**Comment by daxian-dbw** [comment]
Alright :)
@Tadas please see https://github.com/PowerShell/PowerShell/pull/17227#issuecomment-1204207217 for my suggested changes and see if you are fine with that.

**Comment by Tadas** [comment]
Yep the suggested changes look good 👍

**Review by daxian-dbw — APPROVED** [review]
LGTM. Thanks @Tadas!

**Comment by PaulHigin** [comment]
@Tadas  Thank you for your contribution!

**Pull request merged**
